lib/utils.js

```diff
@@ -56,3 +56,13 @@ exports.deepFilterByKey = function (obj, cb) {
     return clean
   }, {})
 }
+
+/**
+ * Given a triple returns a collection of all subjects that occur as initial ' -- '-delimited substrings of the subject
+ * recorded in the triple's object_literal property
+ */
+exports.explodedSubjectLiterals = function (triple) {
+  const subject = triple.object_literal.slice(-1) === '.' ? triple.object_literal.slice(0, -1) : triple.object_literal
+  const componentSubjects = subject.split(' -- ')
+  return componentSubjects.map((subject, i) => componentSubjects.slice(0, i + 1).join(' -- ').trimRight())
```

Comment:
We probably should use trimEnd() but trimRight(). Same thing but looks like trimRight() has some compatibility issues.

http://man.hubwiz.com/docset/JavaScript.docset/Contents/Resources/Documents/developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/TrimRight-2.html

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/trimEnd

Reply:
Thanks I'll change that